base/ltlists.dtx

```diff
@@ -715,7 +715,65 @@
 % \cs{everypar}.
 %
 % This allows paragraph-making environments to work right when called
-% by other environments. (Changed 27 Oct 86)
+% by other environments. (Changed 27 Oct 86).
+%
+% In 2024 this logic was partially replaced with a new algorithm:
+% \begin{itemize}
+% \item
+%   \cs{if@endpe} is now set globally to \texttt{true} or \texttt{false}.
+% \item
+%    In addition \cs{@endpetrue} also initiates an \cs{aftergroup} call
+%    to \cs{propagate@doendpe} if it is called inside a group.
+% \item
+%    \cs{propagate@doendpe} in turn checks the status of \cs{if@endpe}
+%    and if that is \texttt{true} it calls \cs{@doendpe} otherwise it
+%    does nothing.
+% \item
+%    \cs{@doendpe} in turn calls \cs{@endpetrue} and also changes
+%    \cs{par} and \cs{everypar} to do end environment \cs{par}
```

Comment:
not really. it is not about par at the end of the env but about blank lines after the env

Reply:
OK, good: so now I can parse it! 

Try this:

... in turn sets ...true}; and it also makes the necessary changes to ... and ... so that they handle as before any empty line that follows the environment.

Reply:
it doesn't  "set" `\@endpetrue` it "calls" `\@endpetrue` (which in turn sets `\if@endpe` to `\iftrue`--- but this is so basic and standard I'm not going to spell that out)

The other changes I made.